datafusion/functions/src/unicode/reverse.rs

```diff
@@ -83,10 +85,17 @@ impl ScalarUDFImpl for ReverseFunc {
 /// reverse('abcde') = 'edcba'
 /// The implementation uses UTF-8 code points as characters
 pub fn reverse<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
-    let string_array = as_generic_string_array::<T>(&args[0])?;
+    if args[0].data_type() == &Utf8View {
```

Comment:
probably we can keep same pattern like in `invoke` ? it would be more readable

Reply:
as in using a match? Sure. It won't be the same match arms though.